libflux/src/semantic/mod.rs

```diff
@@ -19,3 +19,23 @@ mod parser;
 
 #[cfg(test)]
 mod tests;
+
+use crate::ast;
+use crate::parser;
+use crate::semantic::analyze::Result;
+use crate::semantic::fresh::Fresher;
+
+pub fn analyze_string(source: &str) -> Result<nodes::Package> {
+    let file = parser::parse_string("", source);
+    let errs = ast::check::check(ast::walk::Node::File(&file));
+    if errs.len() > 0 {
+        return Err(format!("got errors on parsing: {:?}", errs));
+    }
+    let ast_pkg = ast::Package {
+        base: file.base.clone(),
+        path: "".to_string(),
+        package: "main".to_string(),
+        files: vec![file],
+    };
+    analyze(ast_pkg, &mut Fresher::new())
```

Comment:
It might be better if we expose a version of the `analyze` function that doesn't take a fresher instance. Like we can change `analyze` to take in only an ast package and then have an `analyze_with` method that takes a fresher instance. That way we make things more composable.

Reply:
Yeah, I see. Anyways, exposing a function that does all the error checking and so on is important. I think that can be done in another PR.

What is weird, is that you could `pub use analyze` without exposing the `Fresher` 🤔

Reply:
(in the meantime) I changed this to not depend on the internal `Result` type but on a generic stringy one

Reply:
That part was ok. I think returning a string is probably going to result in a mistake so please change it back to the result type. I think I am changing my review to say that it is better to expose a version of analyze that doesn't take a fresher instance and one that does.